include/mbedtls/ccm_default.h

```diff
@@ -0,0 +1,132 @@
+/**
+ * \file ccm_default.h
+ *
+ * \brief default function definitions for ccm
+ *
+ *  Copyright (C) 2006-2015, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+#ifndef MBEDTLS_CCM_INTERNAL_H
+#define MBEDTLS_CCM_INTERNAL_H
+#include "cipher.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+* \brief          CCM context structure
+*/
+typedef struct {
+    mbedtls_cipher_context_t cipher_ctx;    /*!< cipher context used */
+} mbedtls_ccm_context_default;
+
+/**
+* \brief           Initialize default CCM context (just makes references valid)
+*                  Makes the context ready for mbedtls_ccm_setkey() or
+*                  mbedtls_ccm_free().
+*
+* \param ctx       default CCM context to initialize
+*/
+void mbedtls_ccm_init_default( mbedtls_ccm_context_default *ctx );
+
```

Comment:
If we're not using `MBEDTLS_CCM_ALT`, the compiler is still forced to export the default-functions and cannot inline them into the main interface functions (at least not without code duplication). What about defining some constant `MBEDTLS_CCM_DEFAULT_INLINE` as `static inline` if `MBEDTLS_CCM_ALT` is not defined, as empty otherwise, and prefixing all default functions with that?